src/db/DbShellCommand.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace craft\db;
+
+use Closure;
+use Composer\Util\Platform;
+use Craft;
+use craft\base\Component;
+use craft\helpers\Db;
+use craft\helpers\FileHelper;
+use craft\helpers\StringHelper;
+use mikehaertl\shellcommand\Command as ShellCommand;
+use PDO;
+use yii\base\ErrorException;
+use yii\base\Exception;
+
+/**
+ * @property-read string $execCommand
+ * @property-read ShellCommand $command
+ */
+abstract class DbShellCommand extends Component
+{
+    public ?Closure $callback = null;
+
+    protected function getCommand(): ShellCommand
+    {
+        $shellCommand = new ShellCommand();
+        $shellCommand->escapeArgs = false;
+
+        return $shellCommand;
+    }
+
+    public function getExecCommand(): string
+    {
+        return $this->getCommand()->getExecCommand();
+    }
+
+    /**
+     * Creates a temporary my.cnf file based on the DB config settings.
+     *
+     * @return string The path to the my.cnf file
+     * @throws ErrorException
+     */
+    protected function createDumpConfigFile(): string
```

Comment:
Worth renaming to `createMySQLDumpConfigFile()` so it's more explicit?

Reply:
Probably? – only reason I didn't is it seems to have logic in it for handling pgsql, oddly enough.

Reply:
Hrm... that feels unnecessary, given the first thing it's going to do is throw an exception if it's not MySQL. Probably some leftover code.

Reply:
Renamed and removed pg conditions